packages/core/src/interactions/network/heartbeat.ts

```diff
@@ -58,59 +59,76 @@ class Heartbeat implements AbstractInteraction {
       // NB. This is a false assumption for 'ping' and we therefore trigger
       // errors.
       let struct: Handler
-      let aborted = false
 
       const abort = new AbortController()
 
       const timeout = setTimeout(() => {
-        aborted = true
         abort.abort()
         verbose(`heartbeat timeout while querying ${counterparty.toB58String()}`)
         reject(Error(`Timeout while querying ${counterparty.toB58String()}.`))
       }, HEARTBEAT_TIMEOUT)
 
+      console.log(`before dialProtocol`)
+
+      console.log(`previous connection`, this.node.connectionManager.connections.get(counterparty.toB58String()))
+
       try {
-        struct = await this.node
-          .dialProtocol(counterparty, this.protocols[0], { signal: abort.signal })
-          .catch(async (err: Error) => {
-            verbose(`heartbeat connection error ${err.name} while dialing ${counterparty.toB58String()} (initial)`)
-            const { id } = await this.node.peerRouting.findPeer(counterparty)
-            //verbose('trying with peer info', peerInfo)
-            return await this.node.dialProtocol(id, this.protocols[0], { signal: abort.signal })
-          })
+        struct = await this.node.dialProtocol(Multiaddr(`/p2p/${counterparty.toB58String()}`), this.protocols[0], {
+          signal: abort.signal
+        })
       } catch (err) {
-        verbose(
-          `heartbeat connection error ${err.name} while dialing ${counterparty.toB58String()} (subsequent)`,
-          aborted
-        )
-        clearTimeout(timeout)
-        if (!aborted) {
-          error(err)
+        if (err.type === 'aborted') {
+          return reject()
         }
+        error(`heartbeat connection error ${err.name} while dialing ${counterparty.toB58String()} (initial)`, err)
+      }
+
+      console.log(`struct`, struct)
+
+      if (abort.signal.aborted) {
         return reject()
       }
 
-      if (aborted) {
-        verbose('aborted but no error')
-        return
+      if (struct == null) {
```

Comment:
No helper for this like we discussed?

Reply:
Not yet. Can only do one thing at one. Part of the helper function is now included in the transport module - where it should live